lib/auth.js

```diff
@@ -0,0 +1,248 @@
+'use strict';
+
+const unicodeCategories = require('./utils/unicode-categories');
+
+const unicodeRangesIncludes = (ranges, codePoint) => {
+  let left = 0;
+  let right = ranges.length - 1;
+  while (left <= right) {
+    const mid = (left + right) >>> 1;
+    const value = ranges[mid];
+    if (typeof value === 'number') {
+      if (codePoint > value) left = mid + 1;
+      else if (codePoint < value) right = mid - 1;
+      else return true;
+    } else if (codePoint > value[1]) {
+      left = mid + 1;
+    } else if (codePoint < value[0]) {
+      right = mid - 1;
+    } else {
+      return true;
+    }
+  }
+  return false;
+};
+
+const stringIncludesChars = (str, ranges, charsNumber) => {
+  let number = 0;
+  const utf16singleUnit = 1 << 16;
+  for (let index = 0; index <= str.length - (charsNumber - number); index++) {
+    const codePoint = str.codePointAt(index);
+    if (codePoint > utf16singleUnit) index++;
+    if (unicodeRangesIncludes(ranges, codePoint)) {
+      if (++number === charsNumber) return true;
+    }
+  }
+  return false;
+};
+
+const passwordTests = {
+  MIN_LENGTH: {
+    test: (password, options) => password.length >= options.minLength,
+    hint: options => 'The password must be longer than or equal to ' +
+      `${options.minLength} chars`,
+    options: { minLength: 10 }
+  },
+  MAX_LENGTH: {
+    test: (password, options) => password.length <= options.maxLength,
+    hint: options => 'The password must be less than or equal to ' +
+      `${options.maxLength} chars`,
+    options: { maxLength: 128 }
+  },
+  MIN_PASSPHRASE_LENGTH: {
+    test: (password, options) => password.length >= options.minLength,
+    hint: options => 'The passphrase must be longer than or equal to ' +
+      `${options.minLength} chars`,
+    options: { minLength: 20 }
+  },
+  MAX_REPEATED_CHARS: {
+    test: (password, options) => {
+      const regexp = new RegExp(`(.)\\1{${options.number},}`);
+      return !regexp.test(password);
+    },
+    hint: options => 'The password can not contain a sequence of more than ' +
+      `${options.number} repeat chars`,
+    options: { number: 2 }
+  },
+  MIN_LOWERCASE_CHARS: {
+    test: (password, option) => (
+      stringIncludesChars(password, unicodeCategories.Ll, option.number)
+    ),
+    hint: options => `The password must contain at least ${options.number} ` +
+      'lowercase chars',
+    options: { number: 1 }
+  },
+  MIN_UPPERCASE_CHARS: {
+    test: (password, options) => (
+      stringIncludesChars(password, unicodeCategories.Lu, options.number)
+    ),
+    hint: options => `The password must contain at least ${options.number} ` +
+      'uppercase chars',
+    options: { number: 1 }
+  },
+  MIN_NUMBERS: {
+    test: (password, options) => {
+      const NUMBERS_RANGE = [[49, 57]];
+      return stringIncludesChars(password, NUMBERS_RANGE, options.number);
+    },
+    hint: options => `The password must contain at least ${options.number} ` +
+      'numbers',
```

Comment:
If the `number` is equal to 1, this will say `The password must contain at least 1 numbers`, which is incorrect. There are similar issues with all of the hints.
These issues would also be solved by enabling i18n since correct pluralization is usually a part of it.